### PR TITLE
HDXDSYS-807 Use better phonetic matcher 

### PR DESCRIPTION
**Pull request by mcarans**
Uses rapidfuzz token sort. I need to test this against a larger set of data. I want to see if it incorrectly matches or fails to match. Any suggestions?

### CODE REVIEW

**Review by turnerm — APPROVED** [review]
One last thing I'm curious about is, do different approaches work better with different languages? Is it even feasible for us to specify the place name languages in the first place? This may be out of scope for now of course.

**Comment by turnerm** [comment]
> Uses rapidfuzz token sort. I need to test this against a larger set of data. I want to see if it incorrectly matches or fails to match. Any suggestions?

Could you generate a dataset? Like take a list of correct admin names and then randomly switch / remove / add characters. Not sure how "realistic" it would be though. You could also ask Chat GPT to imbue it with common typos.

File: src/hdx/location/adminlevel.py [hunk 1]
```diff
@@ -767,7 +769,7 @@ def get_pcode(
                     pcode = name_to_pcode.get(name.lower())
```
[the remark this answers is not in the document]
Comment (mcarans) [reply src/hdx/location/adminlevel.py:769]: It is passed to other functions eg. `self.convert_admin_pcode_length(countryiso3, pcode, **kwargs)`

File: src/hdx/location/adminlevel.py [hunk 2]
```diff
@@ -767,7 +769,7 @@ def get_pcode(
...
                     if pcode:
                         return pcode, True
-            if not fuzzy_match:
+            if not fuzzy_match or len(name) < 4:
```
[the remark this answers is not in the document]
Comment (mcarans) [reply src/hdx/location/adminlevel.py:772]: Good idea!

File: src/hdx/location/names.py [hunk 3]
```diff
@@ -25,5 +25,7 @@ def clean_name(name: str) -> str:
     )
     # Remove all non-ASCII characters
     clean_name = re.sub(non_ascii, " ", clean_name)
+    clean_name = clean_name.replace("'", "")
```
[the remark this answers is not in the document]
Comment (mcarans) [reply src/hdx/location/names.py:28]: Example names look like: Al_Dhale'a. Here we want the _ to be a space but the ' seems to be a part of the name so we would like an end result of al dhalea after cleaning.

File: src/hdx/location/names.py [hunk 4]
```diff
@@ -25,5 +25,7 @@ def clean_name(name: str) -> str:
...
+    clean_name = re.sub(r"[\W_]", " ", clean_name)
     clean_name = unidecode(clean_name)
```
[the remark this answers is not in the document]
Comment (mcarans) [reply src/hdx/location/names.py:30]: Al_Dhale'a and Al Dhale'e / الضالع. In the latter the / makes it through to the  `re.sub(r"[\W_]", " ", clean_name)` line and is replaced there.

File: src/hdx/location/phonetics.py [hunk 5]
```diff
@@ -23,22 +23,22 @@ def match(
...
             alternative_name (str): Alternative name to match. Defaults to None.
             transform_possible_names (ListTuple[Callable]): Functions to transform possible names.
-            threshold: Match threshold. Defaults to 2.
+            threshold (float): Match threshold. Defaults to 53.
 
         Returns:
             Optional[int]: Index of matching name from possible names or None
         """
-        mindistance = None
+        max_similarity = None
```
[the remark this answers is not in the document]
Comment (mcarans) [reply src/hdx/location/phonetics.py:31]: Thanks, that is much neater.

File: src/hdx/location/adminlevel.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (turnerm) [reply src/hdx/location/adminlevel.py:771]: Ah oops sorry :)

**Pull request merged**